swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -106,6 +106,12 @@ private module Cached {
       //or
       // step from previous read to Phi node
       localFlowSsaInput(nodeFrom, def, nodeTo.asDefinition())
+      or
+      // flow through `try!` and similar constructs
+      nodeFrom.asExpr() = nodeTo.asExpr().(AnyTryExpr).getSubExpr()
+      or
+      // flow through `!`
+      nodeFrom.asExpr() = nodeTo.asExpr().(ForceValueExpr).getSubExpr()
```

Comment:
Watch out. This is now a cartesian product with `def`. You should probably move these two cases out of the `exists`.

Reply:
Well spotted.  Fixed.

I've also attempted to comment a couple of the other cases in this predicate that were lacking comments.